### ISSUE TITLE
Skills that default to themselves don't apply correctly

### ISSUE THREAD

**Issue by Lumek4** [richardwilkes/gcs#810]
The skill always defaults to one with a specialization and ignores the one without a specialization.
![image](https://github.com/richardwilkes/gcs/assets/28827788/0b97d0f5-0a33-480d-8c12-04a811858369)

Exception 1: specialized skill won't be defaulted to if it has 0 points (probably directly from rules of no 2-degree defaults).
Exception 2: specialized skill will default to the highest skill correctly if it has 0 points.
![image](https://github.com/richardwilkes/gcs/assets/28827788/eab3d2e9-a4f9-434d-a227-2d3c231e7707)
![image](https://github.com/richardwilkes/gcs/assets/28827788/4a990fb5-e74d-4eb2-bdd4-64296f7be360)
(situation against the rules, only for demonstration)

This can be fixed by adding a "pretend-specialization" to the base skill.
![image](https://github.com/richardwilkes/gcs/assets/28827788/df759d5f-3a18-42df-b958-1dff33925b90)

**Comment by richardwilkes** [comment]
I believe this works correctly now in the latest source base and will be in the next release.

Note that this still requires the data to be setup correctly. For the behavior to be correct for Ritual Magic as defined in the rules, for example, the Basic Set skill Ritual Magic in the current Master Library needs to have its defaults updated from "specialization is anything" to "specialization is" (blank).

@maw3193 Is this change to the Ritual Magic skill (and possibly others with a similar expectation of defaulting) something you can take care of once I make a new release?

**Comment by maw3193** [comment]
Yes, I think so. Though the only skills I can think of that default like this are Ritual Magic and the Optional Specialties (which @rinickolous has tackled in the past in https://github.com/richardwilkes/gcs_master_library/pull/526, so I can start with that for a list.)

**Comment by richardwilkes** [comment]
Agreed. I'm currently working through a number of issues caused by that specialty split, so might be worth being cautious when using the current development version.
